src/Operation/ExecOperation.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Exo\Operation;
+
+use InvalidArgumentException;
+
+class ExecOperation extends AbstractOperation
+{
+    const EXEC = 'execute';
+
+    /**
+     * @var string
+     */
+    private $operation;
+
+    /**
+     * @var string|null
+     */
+    private $body;
+
+    /**
+     * ExecOperation constructor.
+     *
+     * @param string   $name
+     * @param string   $operation
+     * @param ?string   $body
+     */
+    public function __construct(string $name, string $operation, string $body = null) {
+        $this->name = $name;
+        $this->operation = $operation;
+        $this->body = $body;
+    }
+
+    /**
+     * Returns the name.
+     *
+     * @return string
+     */
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    /**
+     * Returns the operation.
+     *
+     * @return string
+     */
+    public function getOperation(): string
+    {
+        return $this->operation;
+    }
+
+    /**
+     * Returns the body.
+     *
+     * @return string
+     */
+    public function getBody(): string
+    {
+        return $this->body;
+    }
+
+    /**
+     * Returns a new operation by applying another operation.
+     *
+     * @param ExecOperation $operation
+     * @return ExecOperation|null
+     */
+    public function apply(ExecOperation $operation)
+    {
+        if ($operation->getName() !== $this->getName()) {
+            throw new InvalidArgumentException('Cannot apply operations for a different execution.');
+        }
+
+        if ($this->getOperation() !== self::EXEC) {
```

Comment:
I think the entire use of this class constant could be avoided since it's the only value used anywhere to represent an `ExecOperation`'s operation.

To that end I would remove the constant along with the `$operation` member variable (and it's constructor argument) so that `ExecOperation::getOperation()` always returns the static string `execute`.

This line would thus be `if ($operation->getOperation() !== $this->getOperation()) {`. At the moment it's checking it's own value and always returns true.

Reply:
With the only operation available being execute, I removed the argument as a whole from both the migration and operation classes, and updated the getOperation() method to return a static string as suggested. Let me know if you agree with the approach or not.